quantum/debounce/sym_eager_pk.c

```diff
@@ -57,13 +57,9 @@ static void transfer_matrix_values(matrix_row_t raw[], matrix_row_t cooked[], ui
 
 // we use num_rows rather than MATRIX_ROWS to support split keyboards
 void debounce_init(uint8_t num_rows) {
-    debounce_counters = (debounce_counter_t *)malloc(num_rows * MATRIX_COLS * sizeof(debounce_counter_t));
-    int i             = 0;
-    for (uint8_t r = 0; r < num_rows; r++) {
-        for (uint8_t c = 0; c < MATRIX_COLS; c++) {
-            debounce_counters[i++] = DEBOUNCE_ELAPSED;
-        }
-    }
+    const size_t debounce_counters_size = num_rows * MATRIX_COLS * sizeof(debounce_counter_t);
+    debounce_counters                   = (debounce_counter_t *)malloc(debounce_counters_size);
+    memset(debounce_counters, DEBOUNCE_ELAPSED, debounce_counters_size);
```

Comment:
This `memset` can easily break if this code gets revisited later on. If the type of `debounce_counter_t` gets upgraded at the same time as the value of `DEBOUNCE_ELAPSED` becomes non-zero, both of which are abstracted in this commit and which a developer will reasonably assume can be changed without much fuss, the counters will then get initialized to the wrong value.
```suggestion
    debounce_counters = (debounce_counter_t *)malloc(num_rows * MATRIX_COLS * sizeof(debounce_counter_t));
    for (size_t i = 0; i <= num_rows * MATRIX_COLS; i++) {
        debounce_counters[i] = DEBOUNCE_ELAPSED;
    }
```